lib/Driver/FrontendUtil.cpp

```diff
@@ -57,6 +58,25 @@ bool swift::driver::getSingleFrontendInvocationFromDriverArguments(
   if (Diags.hadAnyError())
     return true;
 
+  if (ForceNoOutputs) {
+    // Clear existing output modes.
+    ArgList->eraseArg(options::OPT_modes_Group);
+    // Disable other output options that conflict with -typecheck.
+    ArgList->eraseArg(options::OPT_emit_module);
+    ArgList->eraseArg(options::OPT_emit_module_path);
+    ArgList->eraseArg(options::OPT_emit_module_source_info_path);
+    ArgList->eraseArg(options::OPT_emit_module_interface);
+    ArgList->eraseArg(options::OPT_emit_module_interface_path);
+    ArgList->eraseArg(options::OPT_emit_objc_header);
+    ArgList->eraseArg(options::OPT_emit_objc_header_path);
```

Comment:
Does emit_module_doc belong in this group? Or is -typecheck fine with that?

Reply:
This is filtering the driver options and `-emit-module-doc-path` is a frontend-only option. My expectation is that removing `-emit-module` will handle both.  I check this in the driver/createCompilerInvocation test case (which is testing the resulting frontend options).

If you think of any other flags I'm missing though, that would be great - I did an audit to look for compatibility issues, but may have missed something.  That's the principle danger in this PR.